src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/CSharpVirtualDocument.cs

```diff
@@ -47,13 +52,16 @@ public override VirtualDocumentSnapshot Update(IReadOnlyList<TextChange> changes
 
             _hostDocumentSyncVersion = hostDocumentVersion;
 
+            TryRevertProvisionalChanges();
+            _hasProvisionalChanges = provisional;
+
             if (changes.Count == 0)
             {
                 _currentSnapshot = UpdateSnapshot();
                 return _currentSnapshot;
             }
 
-            var edit = TextBuffer.CreateEdit();
+            using var edit = TextBuffer.CreateEdit();
```

Comment:
🎉

Reply:
Ya this is super dope, I've seen several instances where this code path can explode due to dev bugs and then an "active text edit" is till happening behind the scenes which exacerbates the issue.